docs/BR.md

```diff
@@ -1422,53 +1434,31 @@ In all cases, the CA SHALL:
 #### 6.1.1.3 Subscriber Key Pair Generation
 The CA SHALL reject a certificate request if the requested Public Key does not meet the requirements set forth in Sections 6.1.5 and 6.1.6 or if it has a known weak Private Key (such as a Debian weak key, see <http://wiki.debian.org/SSLkeys>).
 
+The CA SHALL reject a certificate request if the CA generated the Key Pair for the Subscriber and the Subscriber Certificate will contain an extendedKeyUsage extension containing either the values id-kp-serverAuth [RFC5280] or anyExtendedKeyUsage [RFC5280].
```

Comment:
This sentence is a bit confusing to me (not a native English speaker). I think it is rejecting something when the CA should not generate in the first place.
Please consider the following alternative language:

> The CA SHALL NOT generate a Key Pair on behalf of a Subscriber, if the associated Subscriber Certificate will contain an extendedKeyUsage extension containing either the values id-kp-serverAuth [RFC5280] or anyExtendedKeyUsage [RFC5280].

Reply:
Thanks @dzacharo for raising this.

While I appreciate your attempt at the language, it introduces a big problem, and so I don't plan to adopt it, but I'm open to finding better wording. This was discussed somewhat in the servercert-wg ( https://archive.cabforum.org/pipermail/servercert-wg/2020-May/001947.html ) and the PR linked from that mail ( https://github.com/sleevi/cabforum-docs/pull/25 )

The scenario this language is trying to address is the one discussed at https://archive.cabforum.org/pipermail/servercert-wg/2020-May/001956.html

Where the CA generates a key for one usage, and the Subscriber later reuses that key for another usage. The language you propose would not address that case. This is requiring that the CA consider the keys they have *historically generated*, and not just the key they are *currently* generating, otherwise, there's a giant loophole.

Reply:
How about:

> The CA SHALL NOT generate or use any Key Pair it generated on behalf of a Subscriber, if the associated Subscriber Certificate will contain an extendedKeyUsage extension containing either the values id-kp-serverAuth [RFC5280] or anyExtendedKeyUsage [RFC5280].

Reply:
```suggestion
The CA SHALL reject a certificate request if the requested Public Key is for a Key Pair that was generated by the CA, either as part of processing this certificate request or any previous certificate request, and the Subscriber Certificate will contain an extendedKeyUsage extension containing either the values id-kp-serverAuth [RFC5280] or anyExtendedKeyUsage [RFC5280].
```

Is this any better?

Reply:
Not really :-(

Reply:
Drat. My worry with your proposed language is that CAs will see "use any Key Pair it generated" as being part of processing the _current_ request, and so I want to find a way to make sure that it's as unambiguous as possible that it applies for all keys it ever generated.

This is also why "SHALL NOT generate" is, I think, a little weird, because "We didn't generate... this time", or, to use ETSI as an example, to generate a key for a non-BR use case like eSeal, and then use that key in a TLS cert. It's not the act of generation that's the problem (CAs can generate as many keys as they want), but using those keys is the problem :(

Reply:
My first attempt was the following:

> The CA SHALL NOT generate and SHALL NOT use any Key Pair it ever generated on behalf of a Subscriber, if the associated Subscriber Certificate will contain an extendedKeyUsage extension containing either the values id-kp-serverAuth [RFC5280] or anyExtendedKeyUsage [RFC5280].

It repeats the "SHALL NOT" and has the "it **ever** generated on behalf of a Subscriber". Check it out and see if it closes the loophole.

Reply:
```suggestion
If the Subscriber Certificate will contain an extendedKeyUsage extension containing either the values id-kp-serverAuth [RFC5280] or anyExtendedKeyUsage [RFC5280], the CA SHALL NOT generate a Key Pair on behalf of a Subscriber, and SHALL NOT accept a certificate request using a Key Pair previously generated by the CA. 
```

What do you think? I'm a little nervous about the "if" hanging at the end, so I restructured it a little to make it clear it applies to both clauses (generate || generated).

Reply:
Yes, this seems much clearer than the first version. Thank you.